lib/cartopy/mpl/geoaxes.py

```diff
@@ -1660,23 +1660,44 @@ def _pcolormesh_patched(self, *args, **kwargs):
             wrap_proj_types = (ccrs._RectangularProjection,
                                ccrs._WarpedRectangularProjection,
                                ccrs.InterruptedGoodeHomolosine,
-                               ccrs.Mercator)
+                               ccrs.Mercator,
+                               ccrs.LambertAzimuthalEqualArea,
+                               ccrs.AzimuthalEquidistant,
+                               ccrs.TransverseMercator,
+                               ccrs.Stereographic)
             if isinstance(t, wrap_proj_types) and \
                     isinstance(self.projection, wrap_proj_types):
 
                 C = C.reshape((Ny - 1, Nx - 1))
                 transformed_pts = transformed_pts.reshape((Ny, Nx, 2))
 
-                # Compute the length of edges in transformed coordinates
+                # Compute the length of diagonals in transformed coordinates
+                # This takes into account the case where the top of the cell
+                # is on one side of the boundary while the other is on the
+                # other side, case overlooked with the edges only.
+                #
+                #    Top----Top  |
```

Comment:
Rather than Top -- Top, what would think about numbering the 4 corners of the quad? Something like the following...

Compute the length of the diagonals in transformed coordinates.
If either diagonal (x0 - x2) or (x1 - x3) is over half the length of the
projection limits it will be masked. This accounts for long edges,
both points of an edge are on the same side of the boundary
and transposed coordinates.
x0 ---- x1
 |            |
 x3 ---- x2

Reply:
ok